### DISCUSSION TITLE
Hide my API Key (network tab

### DISCUSSION THREAD

**Discussion by flipkoehler** [neuefische/questions#433] [Web] [answered]
Hello there, 

I have connected an external API through which I load movies into my app. I received an API key from the provider https://www.themoviedb.org/, which I don't want to share with the world.

I have already managed to store the API key in the env.local file and in Vercel. This also works so far with the reference in my code to the env.local key process.env.NEXT_PUBLIC_xy (see Components / AddAMovie.js), so the API Key is no longer present in my code on Github. However, the key is still viewable when I make a search in the preview (see screenshot - in the network tab). 
![Bildschirmfoto 2022-12-30 um 12 03 44](https://user-images.githubusercontent.com/96984853/210063686-2ecaaf81-63c9-45ba-b000-c2a7f47cddf3.png)

You can reach the [search preview here](https://movieflip-git-feature-add-a-new-movie-flipkoehler.vercel.app/add-a-movie)
and [here is my open PR ](https://github.com/flipkoehler/capstone-project-ts/pull/18)

Do you have any idea how I can make the key "invisible" here as well?

Thanks in advance and best regards

Philip

**Comment by onebarloop** [comment, marked as answer]
Hi Philip. I suppose you could write your own API that communicates with moviedb. 

For example: A POST-request to /api/[nameofthemovie] would trigger your backend to call the moviedb API with your provided key and moviename.

**Comment by flipkoehler** [comment]
Hey Alex, 
thanks for the tip :) 
I had already suspected that tried that in a quick run - unfortunately without success. If I still have time at the end of the capstone phase, I'll try it again!
